mediatek/ramips: disable unsupported background radar detection

DragonBluep opened the pull request:
Devices with MT7905, or MT7915 but lacking a DFS antenna do not support background radar detection, so just disable it.

All OpenWrt supported devices that using MT7905/MT7911/MT7915:

**Already Disabled:**
- [x] h3c,tx1800-plus
- [x] h3c,tx1801-plus
- [x] h3c,tx1806
- [x] tenbay,t-mb5eu-v01
- [x] ubnt,unifi-6-lite

**MT7905:**
- [x] asus,rt-ax53u 8c00fd9b4519bf0ef8fb3470a6df421b9f38c03c
- [x] jcg,q20 https://www.acwifi.net/14024.html
- [x] tplink,eap615-wall-v1 a1b8a4d7b3ff3fa671623e2bccafd43a3c141e07
- [x] xiaomi,mi-router-cr6606 3343ca7e6837b2ac5f237ea78bf73d50831dea20 https://www.acwifi.net/18380.html
- [x] xiaomi,mi-router-cr6608 3343ca7e6837b2ac5f237ea78bf73d50831dea20 same as cr6606/cr6608
- [x] xiaomi,mi-router-cr6609 3343ca7e6837b2ac5f237ea78bf73d50831dea20 https://www.acwifi.net/13821.html
- [x] yuncore,ax820 4891b865380e2b7f32acf0893df9c1ca9db8d4ea

**MT7911/MT7915 without DFS antenna:**
- [x] totolink,x5000r https://www.acwifi.net/14722.html
- [x] cudy,x6 https://forum.openwrt.org/t/add-support-for-cudy-x6-wifi-6-router/98197/13
- [x] mediatek: linksys,e8450 reported by @dangowrt
- [x] mediatek: ruijie,rg-ew3200gx-pro https://www.acwifi.net/14783.html
- [x] mediatek: xiaomi,redmi-router-ax6s https://www.acwifi.net/16612.html

**MT7915 with DFS antenna:**
- [x] iptime,ax2004m reported by @mans0n
- [x] netgear,wax202 https://fccid.io/PY321100518/Internal-Photos/Internal-Photos-5257807.pdf
- [x] mediatek: ubnt,unifi-6-lr reported by @dangowrt

**Not sure, but maintainers can add them individually later:**
**(Usually if the router only has 2 or 4 antennas, then it is probably not supported)**
- [ ] zyxel,nwa50ax
- [ ] mediatek: elecom,wrc-x3200gst3

[MT7905 datasheet](https://oss.komect.com/openhomeres/static/images/chipAndModel/%E7%BB%84%E7%BD%91%E8%8A%AF%E7%89%87/MT7905DAN.pdf)
[MT7915 datasheet](https://edit.wpgdadawant.com/uploads/news_file/blog/2020/2406/tech_files/blog_2406_suggest_other_file.pdf)

dangowrt commented:
This is probably also relevant for mediatek/mt7622 target:
 * UniFi 6 LR **does** have an additional antenna for background radar scanning
 * Linksys E8450 aka. Belkin RT3200 **does not** have an additional antenna for background radar scanning

DragonBluep commented:
I added a new commit for mt7622.

DragonBluep commented:
Some DFS antennas are printed directly on the PCB, such as this device:
![SAM_7824](https://user-images.githubusercontent.com/70847398/185617997-aa22b9e1-168b-4123-bf75-06023abafdcc.jpg)

More info: In the new Filogic 830 platform + MT7976AN/MT7975PN HE160 RF chips, it seems that the extra DFS antenna is no longer needed.

DragonBluep committed:
mediatek: disable unsupported background radar detection

MT7915 requires an additional antenna for background radar scanning.
Disable this feature in the following devices that do not have a
separate DFS antenna:
  linksys,e8450
  ruijie,rg-ew3200gx-pro
  xiaomi,redmi-router-ax6s

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep committed:
ramips: disable unsupported background radar detection

Background radar detection is not supported on devices that
using MT7905, so disable this feature in the following devices:
  asus,rt-ax53u
  jcg,q20
  tplink,eap615-wall-v1
  xiaomi,mi-router-cr6606
  xiaomi,mi-router-cr6608
  xiaomi,mi-router-cr6609
  yuncore,ax820

Devices with MT7915 lacking a DFS antenna also do not support
background DFS:
  totolink,x5000r
  cudy,x6

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

dangowrt commented:
picked. Thank you!

mans0n commented:
FYI iptime ax2004m *does* have an additional internal antenna for DFS.
